docs/design/values.md

```diff
@@ -510,29 +547,51 @@ the provided storage.
-storage they create. The simplest example is the expression after the `=` in a
-local `var` declaration.
-
-The next place where a Carbon expression requires an initializing expression is
-the expression operand to `return` statements. We expand more completely on how
-return statements interact with expressions, values, objects, and storage
-[below](#function-calls-and-returns).
-
-The last path that requires forming an initializing expression in Carbon is when
-attempting to convert a non-reference expression into an ephemeral reference
-expression: the expression is first converted to an initializing expression if
-necessary, and then temporary storage is materialized to act as its output, and
-as the referent of the resulting ephemeral reference expression.
+There are no contexts in Carbon that always require an initializing expression,
+and no expression syntax that always produces an initializing expression. By
+default, function call expressions are initializing expressions, and
+correspondingly the operand of `return` is required to be an initializing
+expression, but this default can be overridden by the
+[function signature](#function-calls-and-returns).
+
+The other case that requires forming an initializing expression in Carbon is
+when attempting to convert an expression into an ephemeral owning reference
+expression (particularly to match a `var` pattern): the expression is first
+converted to an initializing expression if necessary, and then temporary storage
+is materialized to act as its output, and as the referent of the resulting
+ephemeral reference expression.
+
+### Initializing outcomes
+
+An _initializing outcome_ is the result of evaluating an initializing
+expression, and represents an obligation to provide storage for an object of the
+expression's type. This obligation can be fulfilled by allocating suitable
+storage and materializing the initializing outcome into it, or it can be
+delegated by returning it to some enclosing context, where it acts as an
+initializing expression.
+
+The holder of the initializing outcome must fulfill that obligation exactly
+once, and if it is nested within a larger composite outcome (see
+[below](#form-conversions)), the computation that fulfills the obligation must
+be independent of any values or references in the enclosing complete outcome,
+and free of other side-effects. These requirements are enforced during
+typechecking. This ensures that the storage location can actually be computed
+beforehand, and passed as an input to evaluation of the initializing expression.
+
+> **Future work:** At present these requirements are somewhat trivial because an
+> initializing outcome is always fulfilled immediately, but a forthcoming
+> proposal is expected to introduce a way to bind a name to an initializing
+> outcome. That proposal may impose stricter requirements than the ones
+> described here, which are just the minimum necessary for initializing outcomes
+> to be coherent.
```

Comment:
As far as I can tell, you are referring to `let x:? auto = F();`. In the case `F()` returns an initializing expression, `x` becomes a `var`. I don't think #5389 will introduce a way to bind a name to an initializing outcome.

Reply:
What about this?

> `~i` allows you to forward the initializing expression used to initialize `i` when `i` is a `var` parameter.

That sounds to me like `i` is bound to an initializing outcome, not a variable.

Reply:
OK, I left a suggestion on #5389 for rephrasing that sentence, and I've revised this section to focus on initializing outcomes as a specification convenience.